### ISSUE TITLE
Video with channel names without ascii characters fails to download

### ISSUE THREAD

**Issue by N72826** [bbilly1/tubearchivist#146]
first off I want to say, I highly appreciate this project. I am having issues downloading this video in a playlist:
https://www.youtube.com/watch?v=iWkJV_sgYbc
I rescan and it refuses to add it to the queue, so I add it directly to the queue and it still refuses.
I checked logs in portainer, and it is throwing an error for another video by a creator with a blank username probably represented by some unicode character. The thing is, I understand this persons username is probably causing the issue, since it outputs the other videos I have downloaded from them into the main folder instead of creating a directory for that uploader, but all of the other uploads from them downloaded without a problem, and even the one it is pointing to as problematic "20211219_OQnwq3yMGh4_before 2022 watch this.mp4" has already been downloaded into the folder. thank you kindly for this platform and your guidance.
```https://www.youtube.com/watch?v=iWkJV_sgYbc

processing: https://www.youtube.com/watch?v=iWkJV_sgYbc

ParseResult(scheme='https', netloc='www.youtube.com', path='/watch', params='', query='v=iWkJV_sgYbc', fragment='')

[{'url': 'iWkJV_sgYbc', 'type': 'video'}]

[2022-01-12 20:28:35,805: INFO/MainProcess] Task home.tasks.extrac_dl[1d4e7dee-5ca3-4c7a-884b-c94b50518dd7] received

[2022-01-12 20:28:35,855: ERROR/ForkPoolWorker-6] Task home.tasks.extrac_dl[1d4e7dee-5ca3-4c7a-884b-c94b50518dd7] raised unexpected: NotADirectoryError(20, 'Not a directory')

Traceback (most recent call last):

  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 451, in trace_task

    R = retval = fun(*args, **kwargs)

  File "/usr/local/lib/python3.10/site-packages/celery/app/trace.py", line 734, in __protected_call__

    return self.run(*args, **kwargs)

  File "/app/home/tasks.py", line 131, in extrac_dl

    all_videos_added = pending_handler.add_to_pending(missing_videos)

  File "/app/home/src/download.py", line 85, in add_to_pending

    self.all_downloaded = self.get_all_downloaded()

  File "/app/home/src/download.py", line 225, in get_all_downloaded

    videos = os.listdir(channel_path)

NotADirectoryError: [Errno 20] Not a directory: '/youtube/20211219_OQnwq3yMGh4_before 2022 watch this.mp4'```

**Comment by bbilly1** [comment]
So I think there are two different issues at play: 
- OQnwq3yMGh4: That is the same bug as described in #127
- iWkJV_sgYbc: There is no issue with that video but most likely you'll have to first delete the file from the folder that is throwing the `NotADirectoryError` error at the bottom of your traceback, after that you should be able to download this video, also double check the /cache/download folder, to make sure this is empty.

Then until there is a fix for channel names without ascii characters, I'd recommend to remove that channel from your subscribed list for now.

**Comment by N72826** [comment]
thank you man. I highly appreciate it. I had a feeling it wasn't `iWkJV_sgYbc` since the error is regarding `OQnwq3yMGh4` like you said, but I just found it odd that it was capable of downloading the rest of his content without issue.
![image](https://user-images.githubusercontent.com/21070431/149262830-d2fab865-9c6a-4b72-b5fa-25b4d242bad8.png)
Now I know how to troubleshoot it though, thank you again. It's a wonderful thing you have made. 
One last thing I have a couple of questions/suggestions while I have you here, do you plan on adding a feature that reads embedded chapters? I see subtitles are already on your to-do list so that's awesome. I would also like to scan more frequently than once a day but currently I don't know if that is possible given the format, or I am simply not understanding the cron formatting. And when clicking on videos, why does clicking on the title of the video send you to a screen with more detailed information about the video than if you were to click the thumbnail of the video? shouldn't they send you to the same screen?
![image](https://user-images.githubusercontent.com/21070431/149265928-0ed08d2a-47b5-45df-8c9a-fa97a45f89e7.png)
![image](https://user-images.githubusercontent.com/21070431/149265961-265b82cb-ff3d-4e87-b990-a91420da0288.png)
top is clicking on titles and bottom is clicking on thumbnail
this seems intentional but not sure for what purpose.

**Comment by bbilly1** [comment]
So for the chapter, this will need a more sophisticated media player, some more details in #119. Something in my mind, but not sure if and when that will happen. Still looking for somebody with more JS experience to help! 

For a more frequent scheduler setting, I'd recommend to read up a bit about cron and the format it uses, the scheduler here implements a subset of the fields, minutes + hours + days-of-week, there are some more examples in the wiki [here](https://github.com/bbilly1/tubearchivist/wiki/Settings#scheduler-setup) too. Just be careful not to bombard YouTube with requests every minute, that's a sure fire way to get yourself blocked or throttled. Some more examples:
- `0 8,16 *` minute field is at 0, means this will execute at the top of the hour, the hour field reads *8,16* the comma is to chain multiple together, so this will execute at 8am and 4pm, (uses 24h notation) then the day field is the wild card `*`, so every day, so put it together, that will run every day at 8:00 and 16:00. 
- `30 0/12 *` minute is at 30, hour 0/12 means *at every hour divisible by 12* and wild card again for day, so this will run every day at 00:30 and 12:30. 
- Also make sure you have your `TZ` variable set in docker-compose for that to work...

I think you can go from here? So I don't know if and when you'll get blocked by youtube, that's why I made a little precaution that you can't use `*` wildcard for the minute, because that will run every minute.

And yes, the play button, will open an inplace video player without the need to refresh the page, that auto plays, but only has very limited metadata, while the dedicated video page at /video/<video-id>/ will have all the metadata to the video, like description, playlist position, likes, dislikes, channel icon... So hopefully the inplace video player will get playlist position too at some point, but probably not all the metadata you see on the video page to keep things easy to navigate.

**Comment by bbilly1** [comment]
there is a fix for this issue in v0.1.1, delete the affected channels when when you download them again, you'll find the media files in a folder with the channel id.
